### PR TITLE
Crashes table now grabs all register values

### PR DESCRIPTION
**Pull request by PoppySeedPlehzr**

### CODE REVIEW

**Comment by PoppySeedPlehzr** [comment]
This fixes issue #2235

**Comment by osqueryer** [comment]
Build origin/pr/2243/merge (Job: 2868) is good! :+1:

**Comment by osqueryer** [comment]
Build origin/pr/2243/merge (Job: 2869) failed :-1:

File: osquery/tables/system/darwin/crashes.cpp [hunk 1]
```diff
@@ -88,18 +83,26 @@ void readCrashDump(const std::string& app_log, Row& r) {
       continue;
     }
 
-    if (kCrashDumpKeys.count(toks[0]) == 0) {
+    if (kCrashDumpKeys.count(toks[0]) == 0 &&
+        kRegisters.find(toks[0]) == kRegisters.end()) {
```
Comment (zwass) [inline osquery/tables/system/darwin/crashes.cpp]: IIRC, we generally prefer `kRegisters.count(toks[0]) > 0` for existence checking in maps and sets.

File: osquery/tables/system/darwin/crashes.cpp [hunk 2]
```diff
@@ -88,18 +83,26 @@ void readCrashDump(const std::string& app_log, Row& r) {
...
       continue;
     }
 
     // Process and grab all register values
-    if (toks[0] == "rax" || toks[0] == "x0") {
-      std::string reg_str = *it + " " + *(++it);
+    if (kRegisters.find(toks[0]) != kRegisters.end()) {
```
Comment (theopolis) [inline osquery/tables/system/darwin/crashes.cpp]: Same here for @zwass comment about existence checks.

File: osquery/tables/system/darwin/crashes.cpp [hunk 3]
```diff
@@ -88,18 +83,26 @@ void readCrashDump(const std::string& app_log, Row& r) {
...
-      alg::replace_all(reg_str, ": ", ":");
-      alg::replace_all(reg_str, "   ", " ");
+      boost::regex rx_spaces("\\s+");
+      boost::regex rx_spaces_colon(":\\s+");
 
-      r["registers"] = std::move(reg_str);
+      line = boost::regex_replace(line, rx_spaces, " ");
+      line = boost::regex_replace(line, rx_spaces_colon, ":");
+
+      if (r["registers"] == "") {
```
Comment (theopolis) [inline osquery/tables/system/darwin/crashes.cpp]:
```cpp
r["registers"] += (r["registers"].empty()) ? std::move(line) : " " + std::move(line);
```

**Comment by osqueryer** [comment]
Build origin/pr/2243/merge (Job: 2875) is good! :+1:

**Pull request merged**